native/cocos/platform/android/AndroidPlatform.cpp

```diff
@@ -64,7 +65,9 @@ void AndroidPlatform::waitWindowInitialized() {
 int32_t AndroidPlatform::loop() {
     while (_jniNativeGlue->isRunning()) {
         pollEvent();
-        runTask();
+        if (cc::gfx::Device::getInstance()->isDeviceAvailable()) {
+            runTask();
+        }
```

Comment:
It will block game logic, right? As we said, we can not just block game logic, may be we should just cancel rendering.